### PR TITLE
chore(providers): rewrite github provider in typescript

### PR DESCRIPTION
**Pull request by mshd**
I rewrote the provider from js to ts and added the profile types.

I've tested it with two accounts, but please test it further as it's a critical provider.

## ☕️ Reasoning

Embrace typescript

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

Fixes: INSERT_ISSUE_LINK_HERE

## 📌 Resources

- [Contributing guidelines](./CONTRIBUTING.md)
- [Code of conduct](./CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/next-auth/src/providers/github.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    token: "https://github.com/login/oauth/access_token",
+    userinfo: {
+      url: "https://api.github.com/user",
+      async request({ client, tokens }) {
+        // Get base profile
+        const profile = await client.userinfo(tokens.access_token!)
+
+        // If user has email hidden, get their primary email from the GitHub API
+        if (!profile.email) {
+          const emails: GithubEmail[] = await (
```
Comment (mshd) [inline packages/next-auth/src/providers/github.ts:81]: I've added to the types to the email request too.

**Review by balazsorban44 — APPROVED** [review]
Awesome, thanks!

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Linting failed, other than that, looks good!

File: packages/next-auth/src/providers/github.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    authorization: {
+      url: "https://github.com/login/oauth/authorize",
+      params: { scope: "read:user+user:email" },
+    },
+    token: "https://github.com/login/oauth/access_token",
+    userinfo: {
+      url: "https://api.github.com/user",
+      async request({ client, tokens }) {
+        // Get base profile
+        const profile = await client.userinfo(tokens.access_token!)
```
Comment (balazsorban44) [inline packages/next-auth/src/providers/github.ts:77]: https://github.com/nextauthjs/next-auth/runs/7305230206?check_suite_focus=true#step:7:22

This throws as well. I believe it should always be `string` though, so maybe a `// @ts-expect-error` is forgivable here. If not, we can change it to throw an error in the future.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

File: packages/next-auth/src/providers/github.ts (hunk 2 above)
Comment (mshd) [reply packages/next-auth/src/providers/github.ts:78]: I've added the same behavior of https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/facebook.ts

**Review by balazsorban44 — APPROVED** [review]
Thanks!

**Pull request merged**